content/en/blog/_posts/2023-08-23-kubelet-podresources-api-ga.md

```diff
@@ -0,0 +1,128 @@
+to  return information about the assignment of devices to containers.
+The API is used by [multus CNI](https://github.com/k8snetworkplumbingwg/multus-cni) and by
+[GPU monitoring tools](https://github.com/NVIDIA/dcgm-exporter).
+
+Since its inception, the podresources API increased its scope to cover other resource managers than device manager.
+Starting from Kubernetes 1.20, the `List` API reports also CPU cores and memory regions (including hugepages); the API also
+reports the NUMA locality of the devices, while the locality of CPUs and memory can be inferred from the system.
+
+In Kubernetes 1.21, the API [gained](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/2403-pod-resources-allocatable-resources/README.md)
+the `GetAllocatableResources` function.
+This newer API complements the existing `List` API and enables monitoring agents to determine the unallocated resources,
+thus enabling new features built on top of the podresources API like a
+[NUMA-aware scheduler plugin](https://github.com/kubernetes-sigs/scheduler-plugins/blob/master/pkg/noderesourcetopology/README.md).
+
+Finally, in Kubernetes 1.27, another function, `Get` was introduced to be more friendly with CNI meta-plugins, to make it simpler to access resources
+allocated to a specific pod, rather than having to filter through resources for all pods on the node. The `Get` function is currently alpha level.
+
+## Consuming the API
+
+The podresources API is served by the kubelet locally, on the same node on which is running.
+On unix flavors, the endpoint is served over a unix domain socket; the default path is `/var/lib/kubelet/pod-resources/kubelet.sock`.
+On windows, the endpoint is served over a named pipe; the default path is `npipe://\\.\pipe\kubelet-pod-resources`.
+
+In order for the containerized monitoring application consume the API, the socket should be mounted inside the container.
+A good practice is to mount the directory on which the podresources socket endpoint sits rather than the socket directly.
+This will ensure that after a kubelet restart, the containerized monitor application will be able to re-connect to the socket.
+
+An example manifest for a hypothetical monitoring agent consuming the podresources API and deployed as a DaemonSet could look like:
+
+```yaml
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: podresources-monitoring-app
+  namespace: monitoring
+spec:
+  selector:
+    matchLabels:
+      name: podresources-monitoring
+  template:
+    metadata:
+      labels:
+        name: podresources-monitoring
+    spec:
+      containers:
+      - args:
+        - --podresources-socket=unix:///host-podresources/kubelet.sock
+        command:
+        - /bin/podresources-monitor
+        image: podresources-monitor:latest  # just for an example
+        volumeMounts:
+        - mountPath: /host-podresources
+          name: host-podresources
```

Comment:
I see usage of this with `readOnly: true`. Since it is a Directory and not a socket, it should work OK and will limit the exposure. Changing it to readonly will require to add some explanation below explaining the difference between socket non-readonly and directory mount being readonly

Reply:
I'm certainly puzzled about how a read-only mount might let you make a read-write connection to a socket within that mount! That might separately merit an explanation in https://k8s.io/docs/concepts/storage/volumes/#hostpath